Branch-SDK/src/io/branch/referral/ServerRequestRegisterClose.java

```diff
@@ -30,6 +33,18 @@ public ServerRequestRegisterClose(Context context) {
             if (!prefHelper_.getLinkClickID().equals(PrefHelper.NO_STRING_VALUE)) {
                 closePost.put(Defines.Jsonkey.LinkClickID.getKey(), prefHelper_.getLinkClickID());
             }
+            JSONObject ciObject = ContentDiscoverer.getInstance().getContentDiscoverDataForCloseRequest(context);
+            if (ciObject != null) {
+                closePost.put(Defines.Jsonkey.ContentDiscovery.getKey(), ciObject);
+            }
+            try {
+                PackageInfo info = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
```

Comment:
A little worried about garbage collection & activity lifecycle with this code. Can you put this is in your `updateRequestWithDeviceParams` instead?